message-processor/src/main/java/com/bosch/iot/things/example/message/processor/processing/MessageProcessingService.java

```diff
@@ -0,0 +1,28 @@
+package com.bosch.iot.things.example.message.processor.processing;
+
+import org.apache.qpid.proton.amqp.Binary;
+import org.apache.qpid.proton.amqp.messaging.AmqpValue;
+import org.apache.qpid.proton.amqp.messaging.Data;
+import org.apache.qpid.proton.amqp.messaging.Section;
+import org.apache.qpid.proton.message.Message;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MessageProcessingService {
```

Comment:
It makes sense to define an interface for this service, as this will be a custom implementation, depending on the use case. Furthermore we need two methods for both directions (upstream/downstream) to e.g. decrypt/encrypt, right?

Reply:
Sure, that could be done! 
At the time I was just aiming to introduce the message processing in as simple as possible manner and later figure out the details.